connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -167,22 +208,27 @@ export async function populateDeltas(connectorId: ModelId, nodeIds: string[]) {
 
   const client = await getClient(connector.connectionId);
 
-  for (const nodeId of nodeIds) {
-    const node = await MicrosoftNodeResource.fetchByInternalId(
-      connectorId,
-      nodeId
-    );
+  for (const driveId of Object.keys(groupedItems)) {
+    if (groupedItems[driveId]) {
```

Comment:
is the if needed here? not sure there can be undefined keys in groupedItems

Reply:
no it can't, but typescript does not see it